datamap/mv/core/src/main/scala/org/apache/carbondata/mv/rewrite/Utils.scala

```diff
@@ -416,6 +416,30 @@ object Utils extends PredicateHelper {
     }
   }
 
+  private def canBeDerived(subsumer: ModularPlan, expE: Expression): Boolean = {
+    var canBeDerived = false
+    subsumer.asInstanceOf[Select].outputList.forall {
+      case Alias(s: ScalaUDF, _) =>
+        expE.children.foreach { expr =>
+          if (s.semanticEquals(expr)) {
+            canBeDerived = true
```

Comment:
can directly return here and stop checking other expression if found ?

Reply:
there will be one UDF expression only and one reference column inside that, so its ok